projects/ngneat/error-tailor/src/lib/control-error.directive.ts

```diff
@@ -80,8 +82,18 @@ export class ControlErrorsDirective implements OnInit, OnDestroy {
       changesOnBlur$ = blur$.pipe(switchMap(() => valueChanges$.pipe(startWith(true))));
     }
 
-    // submitFirstThenUponChanges
-    const changesOnSubmit$ = this.submit$.pipe(switchMap(() => controlChanges$.pipe(startWith(true))));
+    const submitted$ = merge(this.submit$.pipe(mapTo(true)), this.reset$.pipe(mapTo(false)));
+
+    // when submitted, submitFirstThenUponChanges
+    const changesOnSubmit$ = submitted$.pipe(
+      switchMap(submitted => (submitted ? controlChanges$.pipe(startWith(true)) : NEVER))
+    );
+
+    // on reset, clear ViewContainerRef and ComponentRef
+    this.reset$.pipe(takeUntil(this.destroy)).subscribe(() => {
+      this.anchor.clear();
+      this.ref = null;
```

Comment:
Why not call `ngOnDestroy`?

Reply:
It will completely kill the errors from displaying on successive submits after a reset. Unless i'm misunderstanding the suggestion

Reply:
We should seperate this chunk:

```ts
    if (this.customAnchorDestroyFn) {
      this.customAnchorDestroyFn();
      this.customAnchorDestroyFn = null;
    }
    if (this.ref) this.ref.destroy();
    this.ref = null;
```

and call it in both places.

Reply:
moved logic into new method as suggested, now called from both the reset sub and ngOnDestroy 👍